app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java

```diff
@@ -121,16 +125,66 @@ public void initializeRecyclerView(LinearLayoutManager layoutManager, View view)
             @Override
             // if row at the specified position is clicked
             public void onRowClicked(int position) {
-                openViewWindowForItem(position);
+                // check if a row has been long pressed
+                if (_longPressed) {
+                    // store the to and from positions
+                    int fromPosition = _longPressedPosition;
+                    int toPosition = position;
+
+                    // get the correspondong habits
+                    Habit fromHabit = _habitItemAdapter.getItem(fromPosition);
+                    Habit toHabit = _habitItemAdapter.getItem(toPosition);
+
+                    // change habit item background color back to white
+                    recyclerView.findViewHolderForAdapterPosition(_longPressedPosition).itemView.findViewById(R.id.habit_view_constraint).setBackgroundColor(Color.WHITE);
+
+                    // swap the two habits
+                    swapSortIndex(fromHabit, toHabit, fromPosition, toPosition);
```

Comment:
We should probably have a check to handle the case when a user clicks the same item selected, and thus cancels the operation. I know the code flow right now would lead to the same result but I would prefer we would handle that case more concretely (also so we could maybe display a message indicating it was canceled).

- [ ] Handle case where row clicked is the one that was long pressed